### PR TITLE
Fix ShapeError crash when report type has duplicate fields

### PR DESCRIPTION
**Pull request by Bhavatu**
When a report type had the same attribute type or category type configured twice in its fields StreamField, the Excel export would crash with a Polars ShapeError because columns had mismatched lengths

Changes:
- Add defensive handling in create_populated_actions_dataframe() to detect and skip duplicate fields with a warning log
- Add validation in ReportType.clean() to prevent duplicate fields from being saved in the admin
- Existing bug fix: 'attribute_type' -> 'attribute'
- Add regression tests for duplicate attribute and category fields

Fixes WATCH-BACKEND-3DJ

## Screenshots/Videos (if applicable)
Add screenshots or videos demonstrating the changes if applicable.

## Related issue
https://kausaltech.slack.com/archives/C01CB95BT0C/p1768975260391599?thread_ts=1768902071.216909&cid=C01CB95BT0C

## Requirements, dependencies and related PRs
Describe or link possible requirements, dependencies and related PRs here

## Additional Notes
The original report that caused this didnt exist anymore, so couldn't recreate exactly.

------

### Type of Change
- [x] Set the PR's label to match the nature of this change

### Testing
- [x] **Built Unit tests** (unit tests added/updated)
- [ ] **Built E2E tests** (if applicable. E2E tests added/updated)
- [ ] **Authorization is tested** (permissions and access controls verified)
- [ ] **Manually tested locally** (functionality verified)
    ##### Manual testing instructions
    If feature requires manual testing by reviewer, you can provide instructions here.

### Internationalization & Accessibility
- [ ] **New strings are translatable** (all user-facing text uses i18n)
- [ ] **Accessibility** standards met (WCAG compliance, screen reader support)

### Integrations (if applicable)
If there are model changes to models which use any of the features below, verify the new models work together with the features.
For example, when adding a new model, verify the new model instances are copied when copying a plan.
- [ ] **Moderation** integration implemented
- [ ] **Reporting** integration implemented
- [ ] **Copy plan feature** integration implemented

### Dependencies
- [ ] **Dependencies are merged** (if applicable. If the change depends on other PRs e.g. kausal_common)

### CODE REVIEW

File: reports/models.py [hunk 1]
```diff
@@ -139,6 +139,46 @@ def get_action_list_page(self) -> ActionListPage:
...
+            category_type = field.value['category_type']
+            level = field.value.get('category_level')
+            key = f'{block_name}.{category_type.id}'
+            if level:
+                key = f'{key}.level_{level.id}'
+            return key
+        return block_name
+
+    def clean(self):
+        from django.core.exceptions import ValidationError
```
Comment (tituomin) [inline reports/models.py]: Move to module scope (definitely no risk of circular imports)

File: reports/spreadsheets/excel_report.py [hunk 2]
```diff
@@ -278,6 +278,28 @@ def _prepare_serialized_report_data(self) -> tuple[list[SerializedActionVersion]
     def get_column_labels(self, field_name: str) -> set[str]:
         return self.field_to_column_labels.get(field_name, set())
 
+    def _get_field_unique_key(self, field) -> str:
```
Comment (tituomin) [inline reports/spreadsheets/excel_report.py]: This is literally identical to the function above. You can move it to utils.py or somewhere appropriate and import in the 2 usage sites.

File: reports/spreadsheets/excel_report.py [hunk 3]
```diff
@@ -301,11 +323,21 @@ def append_to_key(key: str, value: ReportCellValue, field_name: str) -> None:
...
                 continue
+            # Generate a unique key for this field to detect duplicates
+            field_key = self._get_field_unique_key(field)
+            if field_key in seen_field_keys:
+                logger.warning(
+                    f"Duplicate field '{field_key}' detected in report type '{self.report.type.name}'. "
+                    "Skipping duplicate to prevent DataFrame column length mismatch."
+                )
+                continue
+            seen_field_keys.add(field_key)
```
Comment (tituomin) [inline reports/spreadsheets/excel_report.py:318]: This part also looks familiar but maybe different enough, maybe not worth extracting now.

**Review by tituomin — APPROVED** [review]
Approved, but maybe remove some duplication. Otherwise excellent!
